### PR TITLE
Override relative path prefix

### PR DESCRIPTION
**Pull request by pimeys**
Usage from JavaScript/N-API:

```
let qe = new QueryEngine(datamodel: dm, config_dir: "/path/to/prisma")
```

### CODE REVIEW

**Review by dpetrick — APPROVED** [review]

File: libs/datamodel/core/src/configuration/datasource.rs [hunk 1]
```diff
@@ -40,6 +43,66 @@ impl Datasource {
...
+                }
+
+                for (k, v) in params.into_iter() {
+                    url.query_pairs_mut().append_pair(&k, &v);
+                }
+
+                self.url.value = url.to_string();
+            }
+        }
+    }
```
Comment (tomhoule) [inline libs/datamodel/core/src/configuration/datasource.rs:105]: I think we need product to be aware of that, in particular with the coming environment management work. This is for sure going to cause bugs in the future.

**Review by tomhoule — APPROVED** [review]
As we discussed previously, for the record I feel the same way as @pimeys about this — this is certain to cause problems down the line. Let's revisit and clean up later though — the desired behaviour needs revision but the implementation is good. Approved.

File: libs/datamodel/core/src/configuration/datasource.rs [hunk 2]
```diff
@@ -40,6 +43,66 @@ impl Datasource {
         let capabilities = self.active_connector.capabilities().clone();
         ConnectorCapabilities::new(capabilities)
     }
+
+    /// Sigh. I'm sorry.
+    ///
```
Comment (janpio) [inline libs/datamodel/core/src/configuration/datasource.rs:48]:
```suggestion
```

File: libs/datamodel/core/src/configuration/datasource.rs [hunk 3]
```diff
@@ -40,6 +43,66 @@ impl Datasource {
...
+    /// only if the path is relative.
+    ///
+    /// The other option would've been to just point in our docs that a path
+    /// in a connection string is relative only from the caller point of view.
+    ///
+    /// Now, there is a promise to change this behavior to follow how everybody
+    /// else is doing it, but I wouldn't hold my breath.
+    ///
+    /// Written by Julius de Bruijn in the holy date of 25.3.2021 for the future
+    /// generations.
```
Comment (janpio) [inline libs/datamodel/core/src/configuration/datasource.rs:61]:
```suggestion
```

**Pull request merged**